src/comestible_inv_listitem.cpp

```diff
@@ -0,0 +1,771 @@
+    }
+
+    int name_startpos = 1;
+    nc_color print_color;
+    //for (size_t i = 0; i < cond.size(); i++) {
+    //  auto c = cond[i];
+    //  set_print_color(print_color, c.second);
+    //  mvwprintz(window, point(i + 1, cur_print_y), print_color, "%c", c.first);
+    //}
+    set_print_color( print_color, menu_color );
+    if( is_selected ) { //fill whole line with color
+        mvwprintz( window, point( 1, cur_print_y ), print_color, "%*s", right_bound, "" );
+    }
+    int max_name_length = right_bound - name_startpos - 1;
+    trim_and_print( window, point( name_startpos, cur_print_y ), max_name_length, print_color, "%s",
+                    item_name );
+}
+
+
+char const *comestible_inv_listitem::set_string_params( nc_color &print_color, int value,
+        bool need_highlight )
+{
+    char const *string_format;
+    if( value > 0 ) {
+        print_color = need_highlight ? c_yellow_green : c_green;
+        string_format = "+%d";
+    } else if( value < 0 ) {
+        print_color = need_highlight ? c_yellow_red : c_red;
+        string_format = "%d";
+    } else {
+        print_color = need_highlight ? c_yellow : c_dark_gray;
+        string_format = "";
+    }
+
+    set_print_color( print_color, print_color );
+    return string_format;
+}
+
+void comestible_inv_listitem::set_print_color( nc_color &retval, nc_color default_col )
+{
+    switch( is_selected ) {
+        case SELECTSTATE_NONE:
+            retval = default_col;
+            break;
+        case SELECTSTATE_SELECTED:
+            retval = hilite( c_white );
+            break;
+        case SELECTSTATE_CATEGORY:
+            retval = c_white_red;
+            break;
+        default:
+            break;
+    }
+}
+
+const islot_comestible &comestible_inv_listitem::get_edible_comestible( player &p,
+        const item &it ) const
+{
+    if( it.is_comestible() && p.can_eat( it ).success() ) {
+        // Ok since can_eat() returns false if is_craft() is true
+        return *it.type->comestible;
+    }
+    static const islot_comestible dummy{};
+    return dummy;
+}
+
+
+std::string comestible_inv_listitem::get_time_left_rounded( player &p, const item &it ) const
+{
+    if( it.is_going_bad() ) {
+        return _( "soon!" );
+    }
+
+    time_duration time_left = 0_turns;
+    const time_duration shelf_life = get_edible_comestible( p, it ).spoils;
+    if( shelf_life > 0_turns ) {
+        const double relative_rot = it.get_relative_rot();
+        time_left = shelf_life - shelf_life * relative_rot;
+
+        // Correct for an estimate that exceeds shelf life -- this happens especially with
+        // fresh items.
+        if( time_left > shelf_life ) {
+            time_left = shelf_life;
+        }
+    }
+
+    //commneted code will give approximate times instead of exact ones, in case we want that
+    //const auto make_result = [this](const time_duration& d, const char* verbose_str,
```

Comment:
Please don't commit commented out code. Either it's useful code and should be run, so it should not be a comment at all. Or it is useless code that should not be there in the first place.